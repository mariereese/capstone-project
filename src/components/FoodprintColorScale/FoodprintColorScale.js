import styled from 'styled-components/macro'

export default function FoodprintColorScale({ pointerPosition }) {
  return (
    <ColorScale>
      <Pointer position={pointerPosition} />
    </ColorScale>
  )
}

const ColorScale = styled.div`
  margin: 10px auto;
  position: relative;
  width: 92%;
  height: 14px;
  border-radius: 41px;
  background: linear-gradient(
    90deg,
    var(--green) 0%,
    var(--yellow) 50%,
    var(--orange) 95%
  );
`

const Pointer = styled.div`
  height: 18px;
  width: 8px;
  position: absolute;
  top: -2px;
  left: ${(props) => props.position}%;
  border-radius: 20px;
  background-color: var(--dark-grey);
`
