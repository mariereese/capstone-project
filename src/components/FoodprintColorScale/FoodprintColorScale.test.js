import FoodprintColorScale from './FoodprintColorScale'
import { render } from '@testing-library/react'

describe('FoodprintColorScale', () => {
  it('renders correctly', () => {
    const { container } = render(<FoodprintColorScale pointerPosition={0} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
