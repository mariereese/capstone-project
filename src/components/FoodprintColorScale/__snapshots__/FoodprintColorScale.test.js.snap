// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoodprintColorScale renders correctly 1`] = `
<div
  class="FoodprintColorScale__ColorScale-sc-1lwjm0x-0 kxLUjA"
>
  <div
    class="FoodprintColorScale__Pointer-sc-1lwjm0x-1 hyNTtu"
  />
</div>
`;
